lib/Normalize_std_ast.ml

```diff
@@ -12,6 +12,46 @@
 open Parser_standard
 open Std_ast
 
+let make_attr_with_name name =
+  Ast_helper.Attr.mk ~loc:Location.none
+    (Location.mkloc name Location.none)
+    (PStr [])
+
+let convert_legacy_jane_street_local_annotations ~ghost ?segment =
+  let segment_str =
+    Option.value_map ~default:"" ~f:(fun s -> "." ^ s) segment
+  in
+  let attrs name =
+    List.map ~f:make_attr_with_name
+      [ "jane.erasable.local"
+      ; ( "jane.erasable.local._location."
+        ^ if ghost then "_ghost" else "_nonghost" )
+      ; "jane.erasable.local" ^ segment_str ^ "." ^ name ]
+  in
+  List.concat_map ~f:(fun attr ->
+      match attr with
+      | {attr_name= {txt= old_name; _}; attr_payload= PStr []; _} ->
+          if Conf.is_jane_street_local_annotation "local" ~test:old_name then
+            attrs "local"
+          else if
+            Conf.is_jane_street_local_annotation "global" ~test:old_name
+          then attrs "global"
+          else if
+            Conf.is_jane_street_local_annotation "exclave" ~test:old_name
+          then attrs "exclave"
+          else [attr]
+      | _ -> [attr] )
+
+let extract_legacy_jane_street_local_annotations :
```

Comment:
I fail see the exact issue here.

`List.partition_tf` returns two list, one for things that evaluates to true and one for things that evaluates to false (in this case, it separates out the local attributes from the rest).

Something like `type t = (string[@aaa][@local][@bbb][@ccc]) -> (string[@local])` is currently valid code accepted by the compiler, and ocamlformat correctly transforms it to `type t = local_ (string[@aaa] [@bbb] [@ccc]) -> local_ string` (added as a testcase).

And the `extract_legacy_jane_street_local_annotations` function is only used in one place to move the local attributes from label_declaration into its `pld_type` to align the behavior with the new syntax.

If we are really concerned about attribute ordering, it will have to be a large change since the ast normalization logic currently sorts the attributes with a call to `sort_attributes` and ignores reordering.

Reply:
Yes, it's fine. I didn't notice that this function is used only on label declarations. Sorry for the noise.